### PR TITLE
Adding support for MemAvailable

### PR DESCRIPTION
**Pull request by truthbk**
Based on: https://github.com/DataDog/dd-agent/pull/1826 - bringing in @jraede changes and making sure it's only reported if the kernel actually has it in `/proc/meminfo`

### CODE REVIEW

**Comment by remh** [comment]
This will need some changes in the backend as well. 
@olivielpeau i think you've done similar things in the past when you add guest cpu metrics. Can you work with @truthbk on that please ?

**Comment by truthbk** [comment]
I was adding the MemAvailable stuff to the backend, but it kind of feels like overkill. Maybe we should just report MemAvailable as MemUsable when its available, and compute usable when it's not available (aka. like it's done now). Thoughts?

**Comment by truthbk** [comment]
It doesn't make sense to report usable and available. We just use available if the kernel has it, or compute usable if it's not.

File: checks/system/unix.py [hunk 1]
```diff
@@ -361,13 +365,17 @@ def check(self, agentConfig):
             try:
                 memData['physTotal'] = int(meminfo.get('MemTotal', 0)) / 1024
                 memData['physFree'] = int(meminfo.get('MemFree', 0)) / 1024
+                if 'MemAvailable' in meminfo:
+                    memData['physUsable'] = int(meminfo.get('MemAvailable', 0)) / 1024
+                else:
+                    # Usable is relative since cached and buffers are actually used to speed things up.
+                    memData['physUsable'] = memData['physFree'] + memData['physBuffers'] + memData['physCached']
```
Comment (talwai) [inline checks/system/unix.py]: doesn't look like `physBuffers` and `physCached` are populated until L374-375 . Maybe move this block of code below that?

Comment (truthbk) [inline checks/system/unix.py]: Thanks for spotting that! Careless.

**Comment by talwai** [comment]
:+1:

**Comment by remh** [comment]
@talwai @truthbk Did you check if those metrics were roughly equivalent ? We need to make sure that they'll be concistent with the previous value.

**Comment by truthbk** [comment]
@remh the thing is MemAvailable was introduced precisely because it's not always so straight-forward to estimate the amount of memory available to the system: https://access.redhat.com/solutions/776393. As it turns out the typical way of doing `free + cached + buffers` didn't always reflect the actual memory available.

**Comment by JohnLZeller** [comment]
@remh can +1 what @truthbk is saying here. Looks like the MemAvailable number is lower than our calculation (ex `memavail: 529  VS  calculated: 663`), meaning that our manual calculation is actually off a bit.

**Pull request merged**